css-view-transitions-1/Overview.bs

```diff
@@ -1663,15 +1659,19 @@ urlPrefix: https://wicg.github.io/navigation-api/; type: interface;
 			1. [=list/Append=] |transition| to |transition|'s [=relevant settings object=]'s [=update callback queue=].
 			1. [=Queue a global task=] on the [=DOM manipulation task source=],
 				given |transition|'s [=relevant global object=], to [=flush the update callback queue=].
-
 	</div>
 	<div algorithm>
-		To <dfn>flush the update callback queue</dfn> given a {{Document}} |document|:
+		To <dfn export>flush the update callback queue</dfn> given a {{Document}} |document|:
 			1. [=list/For each=] |transition| in |document|'s [=update callback queue=], [=call the update callback=] given |transition|.
 
 			1. Set |document|'s [=update callback queue=] to an empty list.
 	</div>
 
+	Note: a scheduled update callback is guaranteed to be called before the next rendering steps, regardless of whether the
```

Comment:
So... what's the behavior if you put stuff in the queue in a `requestAnimationFrame` callback with your proposed behavior? That would have a frame delay right? Not necessarily a deal breaker, just trying to understand the implications of this.

Reply:
> So... what's the behavior if you put stuff in the queue in a `requestAnimationFrame` callback with your proposed behavior? That would have a frame delay right? Not necessarily a deal breaker, just trying to understand the implications of this.

I don't think this scenario is valid.

If you have a scheduled update callback that wasn't cancelled yet, meaning that this is an active transition waiting for the new state to settle, rendering is guaranteed to be suppressed so we won't enter the rendering loop.

The only time this would take effect is if there are skipped transitions plus a new one that was scheduled before this rendering loop and hadn't had its old state captured yet.